org.eclipse.jdt.core.tests.model/workspace/gh432MethodReferencesSearchBug/.classpath

```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.MODULE_PATH"/>
```

Comment:
What is that and why is this needed?

Reply:
> What is that and why is this needed?

https://github.com/eclipse-jdt/eclipse.jdt.core/pull/433#discussion_r985627128

Reply:
Doesn't answer first part of the question. Also I wonder how other tests work that have not checked in .classpath files.

Reply:
> Doesn't answer first part of the question. Also I wonder how other tests work that have not checked in .classpath files.

Did you read the comment? **That** snippet doesn't work without a JRE on the classpath, which is normally not added during tests. The rest of the tests don't run into the problem and so they work.

Reply:
OK, you didn't understand my question. 
1) What is **this**: <classpathentry kind="con" path="org.eclipse.jdt.MODULE_PATH"/> nad why it is needed
2) what is special with the concrete test snippet, three other snippets work fine?

Reply:
No idea about `org.eclipse.jdt.MODULE_PATH`, I believe I either copied the `.classpath` of the generated test project and added a JRE to it, or I copied the `.classpath` from another test and added the JRE to it. I can't remember.

> what is special with the concrete test snippet

It has a lambda, and in the lambda body there is an expression that cannot be resolved. The full analysis is listed here: https://github.com/eclipse-jdt/eclipse.jdt.core/pull/433#discussion_r985540992

Reply:
Note: the patch still doesn't work on Windows, since you've ignored all my changes :-(
And `org.eclipse.jdt.MODULE_PATH` isn't needed, but I assume the `.classpath` is also a hack and can be replaced by a proper JDT core test framework usage.

Reply:
> Note: the patch still doesn't work on Windows, since you've ignored all my changes :-( And `org.eclipse.jdt.MODULE_PATH` isn't needed, but I assume the `.classpath` is also a hack and can be replaced by a proper JDT core test framework usage.

Adding a JRE to the classpath will be a no small amount of extra code. As already said in the linked comment, adding a `.classpath` is the easiest option. What is the problem with it?

What change did I ignore? The only change is to this line:

`IJavaProject p = setupModuleProject(testProjectName, new String[] {"src"}, new String[0], null);`

Since for whatever reason this line doesn't import the project.

Reply:
See my last push. That works on Windows & Java 11

Reply:
> See my last push. That works on Windows & Java 11

Works also on Linux, with Java 11 and Java 17.